subprojects/packagefiles/openssl/meson.build.tmpl

```diff
@@ -0,0 +1,97 @@
+# OpenSSL library
+libcrypto_sources = [
+%%-
+foreach $src (@libcrypto_srcs) {
+  $OUT .= "  '$src',\n";
+}
+foreach $src (@generated_srcs) {
+  $OUT .= "  'generated-config/archs/$arch/$asm/$src',\n";
+}
+-%%]
+libssl_sources = [
+%%-
+foreach $src (@libssl_srcs) {
+  $OUT .= "  '$src',\n";
+}
+-%%]
+openssl_defines = [
+%%-
+foreach $define (@{$config{defines}}) {
+  $OUT .= "  '$define',\n";
+}
+foreach $define (@lib_cppflags) {
+  $OUT .= "  '$define',\n";
+}
+foreach $define (@{$target{defines}}) {
+  $OUT .= "  '$define',\n";
+}
+foreach $define (@{$config{lib_defines}}) {
+  $OUT .= "  '$define',\n";
+}
+-%%]
+openssl_cflags = [
+%%-
+my %cflags_used;
+if (!$is_win) {
+  foreach $cflag (@cflags) {
+    foreach $single_cflag (split / /, $cflag) {
+      # Warning and optimization levels are configured on Meson level now, threads will be enabled separately below
+      if (
+        $single_cflag ne '-Wall' &&
+        $single_cflag ne '-O3' &&
+        $single_cflag ne '-pthread' &&
+        !exists($cflags_used{$single_cflag} )
+      ) {
+        $OUT .= "  '$single_cflag',\n";
+        $cflags_used{$single_cflag} = true;
+      }
+    }
+  }
+}
+-%%]
+openssl_libraries = [
+%%-
+foreach $ex_lib (split / /, $target{ex_libs}) {
+  # Threads will be enabled separately below
+  if ($ex_lib ne '-pthread') {
+    # *nix
+    if (substr($ex_lib, 0, 2) eq '-l') {
+      $ex_lib = substr($ex_lib, 2);
+      $OUT .= "  '$ex_lib',\n";
+    }
+    # Windows
+    if ($ex_lib =~ /\.lib$/) {
+      $ex_lib = substr($ex_lib, 0, length($ex_lib) - 4);
+      $OUT .= "  '$ex_lib',\n";
+    }
+  }
+}
+-%%]
+openssl_dependencies = [
+%%-
+foreach $ex_lib (split / /, $target{ex_libs}) {
+  # Convert `-pthread` to portable dependency
+  if ($ex_lib eq '-pthread') {
```

Comment:
When I suggested to remove this from `openssl_cflags`, I meant to store it in meson.build, not meson.build.tmpl for the per-arch subdirectories.

It is currently invoked on every platform where -pthread is relevant, and openssl has a global perl ./Configure option to disable threads. I guess the meson wrap should also have a "threads" option, which toggles whether to add dependency('threads') *once* for all arches.

Reply:
Even when threads are enabled apparently not all platforms need `-pthread` or similar flag. Hence `threads` dependency is only added when `./Configure` generated it. I didn't check what difference it makes between configuring OpenSSL with threads or without, I assume some generated files may also change or some defines to be added/removed.

I would prefer to get MVP of OpenSSL wrap going and then start porting more of the configuration options to Meson.

Reply:
My point is, primarily, that "add the threads dependency" is something safely done in the root meson.build; it will only actually add -pthread on unixy compilers.

The generated-configs/ is using an openssl build configured with threads enabled, so threads is needed on all platforms, not as a per-platform tuneable.

It just happens that -pthread is a per-platform implementation of enabling threads, but using the meson dependency means meson cares about that, and openssl doesn't need to care about that.

Feel free not to include a meson option for threads, but please do follow the advice of the first paragraph. :p